jenkins/publish-snapshot.jenkinsfile

```diff
@@ -0,0 +1,25 @@
+pipeline {
```

Comment:
How does the Jenkins fleet pick this job/pipeline ? 
Is there a corresponding file in opensearch-ci? 

Also I see for every component, jenkins files are defined in `opensearch-build`. Why is this different?

Reply:
@gaiksaya @peterzhuamazon ? 

i think the job needs to be added manually once, but I could be wrong (I hope so)

Reply:
We could move these to OpenSearch-build, I have no strong preference, I was replacing a file that was already here

Reply:
Yeah for now we to manually added files, for releasing opensearch-java it already exists. @saratvemulapalli we are planning to move respective jenkins file for clients in their repo. Would be easier to manage. However, need to add regression tests for all jenkins file to see the behavior is as expected.

Reply:
thanks that helps :).
One concern having these files in individual repositories is that the responsibility is delegated to the maintainers of the individual projects and make sure the infrastructure is not mis-used.

Reply:
I believe the maintenance of these files will be delegated to respective maintainers of the project rather than @opensearch-project/engineering-effectiveness being the bottle neck for everything. Regarding the mis-use of the infrastructure, we have to be careful what triggers the workflows. For example, if a commit push is gonna trigger the workflow no matter where the workflow lies it will be executed.